plugins/node/opentelemetry-instrumentation-cassandra/src/instrumentation.ts

```diff
@@ -0,0 +1,366 @@
+import { VERSION } from './version';
+import { EventEmitter } from 'events';
+import type * as CassandraDriver from 'cassandra-driver';
+
+const supportedVersions = ['>=4.4 <5.0'];
+
+export class CassandraDriverInstrumentation extends InstrumentationBase {
+  constructor(config: CassandraDriverInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-cassandra-driver', VERSION, config);
+  }
+
+  protected init() {
+    return new InstrumentationNodeModuleDefinition<typeof CassandraDriver>(
+      'cassandra-driver',
+      supportedVersions,
+      driverModule => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const Client = driverModule.Client.prototype as any;
+
+        if (isWrapped(Client['_execute'])) {
+          this._unwrap(Client, '_execute');
+        }
+
+        if (isWrapped(Client.batch)) {
+          this._unwrap(Client, 'batch');
+        }
+
+        if (isWrapped(Client.stream)) {
+          this._unwrap(Client, 'stream');
+        }
+
+        this._wrap(Client, '_execute', this._getPatchedExecute());
+        this._wrap(Client, 'batch', this._getPatchedBatch());
+        this._wrap(Client, 'stream', this._getPatchedStream());
+
+        return driverModule;
+      },
+      driverModule => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const Client = driverModule.Client.prototype as any;
+
+        if (isWrapped(Client['_execute'])) {
+          this._unwrap(Client, '_execute');
+        }
+
+        if (isWrapped(Client.batch)) {
+          this._unwrap(Client, 'batch');
+        }
+
+        if (isWrapped(Client.stream)) {
+          this._unwrap(Client, 'stream');
+        }
+      },
+      [
+        new InstrumentationNodeModuleFile(
+          'cassandra-driver/lib/request-execution.js',
+          supportedVersions,
+          execution => {
+            if (isWrapped(execution.prototype['_sendOnConnection'])) {
+              this._unwrap(execution.prototype, '_sendOnConnection');
+            }
+
+            this._wrap(
+              execution.prototype,
+              '_sendOnConnection',
+              this._getPatchedSendOnConnection()
+            );
+            return execution;
+          },
+          execution => {
+            if (execution === undefined) return;
+            this._unwrap(execution.prototype, '_sendOnConnection');
+          }
+        ),
+      ]
+    );
+  }
+
+  public getMaxQueryLength(): number {
+    const config = this.getConfig() as CassandraDriverInstrumentationConfig;
+    return config.maxQueryLength ?? 65536;
+  }
+
+  private _getPatchedExecute() {
+    return (
+      original: (...args: unknown[]) => Promise<CassandraDriver.types.ResultSet>
+    ) => {
+      const plugin = this;
+      return function patchedExecute(
+        this: CassandraDriver.Client,
+        ...args: unknown[]
+      ) {
+        const span = startSpan(
+          plugin.tracer,
+          'execute',
```

Comment:
i think the span name should include that its a span for cassandra ? What about `cassandra.execute` ?